src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js

```diff
@@ -1,56 +1,308 @@
+                }
+
+                return this.childIndices;
+            }
+
+            if (visibleWidth + nextWidth > this.wrapperWidth) {
+                break;
+            }
+
+            visibleWidth += nextWidth;
+            visibleTabIndices = [...visibleTabIndices, index];
+        }
+
+        visibleTabIndices.sort((a, b) => a - b);
+
+        return visibleTabIndices;
+    }
+
+    get visibleTabs() {
+        const {children} = this.props;
+        const visibleTabIndices = this.visibleTabIndices;
+
+        return this.createTabItems(
+            React.Children.toArray(children).filter(
+                (child, index) => visibleTabIndices.includes(index)
+            ),
+            visibleTabIndices
+        );
+    }
+
+    @computed get hiddenTabIndices(): number[] {
+        const visibleTabIndices = this.visibleTabIndices;
+
+        return this.childIndices.filter((index) => !visibleTabIndices.includes(index));
+    }
+
+    @computed get hasHiddenTabs(): boolean {
+        return this.hiddenTabIndices.length > 0;
+    }
+
+    get hiddenTabs() {
+        const {children} = this.props;
+        const hiddenTabIndices = this.hiddenTabIndices;
+
+        return this.createCollapsedTabItems(
+            React.Children.toArray(children).filter((child, index) => hiddenTabIndices.includes(index)),
+            hiddenTabIndices
+        );
+    }
+
+    createTabItems(tabs: Array<Element<typeof Tab> | false>, indices: number[]) {
+        const {small} = this.props;
+
+        return React.Children.map(tabs, (tab, localIndex) => {
+            const index = indices[localIndex];
+
             if (!tab) {
                 return null;
             }
 
+            const selected = this.isSelected(index);
+
             return React.cloneElement(
                 tab,
                 {
                     ...tab.props,
                     index,
-                    selected: this.isSelected(index),
-                    onClick: this.handleSelectionChange,
+                    selected,
+                    small,
+                    onClick: this.handleTabClick,
+                    setWidth: this.setTabWidth,
                 }
             );
         });
     }
 
-    handleSelectionChange = (selectedTabIndex: ?number) => {
-        if (typeof(selectedTabIndex) !== 'undefined' && selectedTabIndex !== null) {
-            this.props.onSelect(selectedTabIndex);
-        }
-    };
+    createCollapsedTabItems(tabs: Array<Element<typeof Tab> | false>, indices: number[]) {
+        const {skin} = this.props;
+
+        return React.Children.map(tabs, (tab, localIndex) => {
+            const index = indices[localIndex];
+
+            if (!tab) {
+                return null;
+            }
+
+            return (
+                <CollapsedTab
+                    index={index}
+                    key={index}
+                    onClick={this.handleCollapsedTabClick}
+                    skin={tab.props.skin || skin}
```

Comment:
I don't really know how. I guess the `index` prop of the `TabComponent` would be the same case, and that is not marked somehow, it's just declared as an optional prop, which it isn't really.

Reply:
Ok, made a quick search, and couldn't find an example either... But I think what made me trip here, is that in this case you only fallback to the value passed to the `Tabs` component... I would say it must always be the value from the `Tabs` component, so this line should look like this:

```javascript
skin={skin}
```

Otherwise you could have tabs with a dfiferent skin in the same tabs bar, and that should not happen, should it?